dubbo-metadata-report/dubbo-metadata-report-api/src/main/java/org/apache/dubbo/metadata/integration/MetadataReportService.java

```diff
@@ -94,7 +94,7 @@ public void publishProvider(URL providerUrl) throws RpcException {
         try {
             String interfaceName = providerUrl.getParameter(INTERFACE_KEY);
             if (StringUtils.isNotEmpty(interfaceName)) {
-                Class interfaceClass = Class.forName(interfaceName);
+                Class interfaceClass =Thread.currentThread().getContextClassLoader().loadClass(interfaceName);
```

Comment:
I think this change is not appropriate, the Thread.currentThread().getContextClassLoader().loadClass may cause NullPointerException if calling Thread.currentThread().setContextClassLoader(null) first.

Reply:
I want to add a context loader to compare whether the same loader is used with the passed interface class. What should I do?

Reply:
I think it should be like the following
```java
ClassLoader classLoader = null == Thread.currentThread().getContextClassLoader() ? this.getClass().getClassLoader() : Thread.currentThread().getContextClassLoader();
 Class interfaceClass = Class.forName(interfaceName, false, classLoader);
```

Reply:
I think it should be like this:
Only when the Class in the same class loader is compared using ==, this is the time when the user registers the interfaceName. , the class loader to which the interfaceName belongs is the same as when it was called.
```
private static boolean isInterfaceAllowed(Class<?> tClass, ClassLoader classLoader) {
        boolean result = false;
        Class<?> aClass = null;
        try {
            // The incoming classLoader is the current class loader that calls interfaceName, looking for the class object of interfaceName
            aClass =  Class.forName(tClass.getName(), true, classLoader);
        } catch (Exception ex) {
            result = false;
        }
        //Only when the Class in the same class loader is compared using ==, this is the time when the user registers the interfaceName.
        //,the class loader to which the interfaceName belongs is the same as when it was called.
        result = ( aClass == tClass ) ? true : false;


         return result;
    }
```